controllers/imageclusterinstall_controller.go

```diff
@@ -227,13 +226,6 @@ func (r *ImageClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 		}
 	}
 
-	if ici.Status.BareMetalHostRef != nil && !v1alpha1.BMHRefsMatch(ici.Spec.BareMetalHostRef, ici.Status.BareMetalHostRef) {
```

Comment:
why? I am not sure we need it too much but still, the idea is that you can sort of change bmh and will be nice to cleanup the old one but we can drop it i believe